src/app/api/tasks/[id]/route.ts

```diff
@@ -0,0 +1,55 @@
+import { NextRequest, NextResponse } from 'next/server';
+import prisma from '../../../../../lib/prismadb';
+
+
+export async function PUT(request: NextRequest, context: { params: NodeJS.Dict<string>  }) {
+  const id = context.params.id;
+  const { title, description, assigneeEmail } = await request.json();
+  const task = await prisma.$transaction(async () => {
+    const task = await prisma.task.update({
+      where: { id },
+      data: {
+        title,
+        description,
+      },
+    });
+    try {
+      assigneeEmail &&
+        (await prisma.assignedTask.update({
+          where: { taskId_assigneeEmail: { taskId: task.id, assigneeEmail } },
+          data: {
+            assigneeEmail,
+            taskId: task.id,
+          },
+        }));
+      return task;
+    } catch (error) {
+      throw error;
+    }
+  });
+  return NextResponse.json(task);
+}
+
+export async function DELETE(request: NextRequest, context: { params: NodeJS.Dict<string> }) {
+  try {
+    const id = context.params.id;
+    await prisma.assignedTask.deleteMany({ where: { taskId: id } });
```

Comment:
Use prisms.$transaction and pass both commands without the await in an array instead

Reply:
Done